src/Utilities/AnsiBuilder.cs

```diff
@@ -0,0 +1,330 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace WheelMUD.Utilities
+{
+    /// <summary>
+    /// Mutable String class for Ansi based strings, optimized for speed and memory while retrieving the final result
+    /// as a string. Similar use to StringBuilder, but avoid a lot of allocations done by StringBuilder
+    /// </summary>
+    public class AnsiBuilder
+    {
+        /// <summary>
+        ///     Working mutable string
+        /// </summary>
+        private char[] buffer;
+
+        private int bufferPos;
+        private int charsCapacity;
+
+        /// <summary>
+        /// Temporary string used for the Replace method
+        /// </summary>
+        private List<char> replacement;
+
+        /// <summary>
+        /// Immutable string. Generated at last moment, only if needed
+        /// </summary>
+        private string stringGenerated = "";
```

Comment:
This member seems only used in one place, and seems unnecessary?

Reply:
This has to do with performance. It doesn't actually generate any string until ToString is processed. This is one of the huge differences between this and StringBuilder. StringBuilder generates the string with each append, this allows it to do more things like Format,Insert, etc, but it comes at a larger cost and allows for code to be sloppier. With the Ansi builder we are forcing good practice (for the most part).

Reply:
What I mean is we can delete this line because the only reference to `stringGenerated` is:
```
        public override string ToString()
        {
            if (needsParsing) Parse();
            stringGenerated = new string(buffer, 0, bufferPos);
            return stringGenerated;
        }
```
which I think we can replace with
```
        public override string ToString()
        {
            if (needsParsing) Parse();
            return new string(buffer, 0, bufferPos);
        }
```
unless you meant to avoid the `new` for the rare case one calls ab.ToString more than once, in which case you're missing some logic to honor the cached stringGenerated without doing the extra work again.